File: java/core/src/main/java/com/github/fernthedev/core/encryption/codecs/json/FastJSONHandler.java
package com.github.fernthedev.core.encryption.codecs.json;

import com.alibaba.fastjson.JSON;
import com.github.fernthedev.core.api.APIUsage;
import com.github.fernthedev.core.encryption.codecs.JSONHandler;

@APIUsage
public class FastJSONHandler implements JSONHandler {
    @Override
    public <T> T fromJson(String json, Class<T> classOfT) {
        return JSON.parseObject(json, classOfT);
    }

    @Override
    public String toJson(Object src) {
        return JSON.toJSONString(src);
    }
}

File: java/core/src/main/java/com/github/fernthedev/lightchat/core/codecs/json/FastJSONHandler.kt
package com.github.fernthedev.lightchat.core.codecs.json

import com.alibaba.fastjson.JSON
import com.github.fernthedev.lightchat.core.api.APIUsage
import com.github.fernthedev.lightchat.core.codecs.JSONHandler

@APIUsage
class FastJSONHandler : JSONHandler {

    override fun <T> fromJson(decodedStr: String, packetWrapperClass: Class<T>): T {
        return JSON.parseObject(decodedStr, packetWrapperClass)
    }

    override fun toJson(src: Any?): String {
        return JSON.toJSONString(src)
    }
}